### ISSUE TITLE
HttpException did not catched within Browserkit/Client->request

### ISSUE THREAD

**Issue by dominikzogg** [symfony/symfony#27720]
**Symfony version(s) affected**:3.4.12

**Description**  
After updating from symfony 3.4.11 => 3.4.12 many of our integration tests break, cause Symfony\Component\HttpKernel\Exception get not catched anymore. Instead of a response with 400 in case of BadRequestHttpException, the test failed cause this exception.

**Comment by xabbuh** [comment]
Can you create a minimal example application that allows to reproduce your issue?

**Comment by nicolas-grekas** [comment]
Related to #27516? This was a BC break that has been reverted (see l linked issue.)

**Comment by dominikzogg** [comment]
https://github.com/dominikzogg/symfony-bug-hunting/pull/1

**Comment by dominikzogg** [comment]
@nicolas-grekas seems so, cause i had the error after removing some bundles (twig and security are two of them)

**Comment by nicolas-grekas** [comment]
This means you should fix your code I suppose, "not a bug"...

**Comment by dominikzogg** [comment]
@nicolas-grekas well, i used a original standard edition for demo, dropped twig, and this means all kernel exceptions need to catch by ourself?

**Comment by nicolas-grekas** [comment]
Yes, that's always been the case except in a few 3.4/4.0 versions but that introduced a BC break. You need to register an exception listener (what TwigBundle does) or to catch the exception yourself.

**Comment by dominikzogg** [comment]
Can you explain me, what http exception handling has to do with twig?

**Comment by nicolas-grekas** [comment]
TwigBundle, not Twig only. See https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml#L140

**Comment by RonnieRocket147** [comment]
I have the same issue after upgrading from 3.4.11 to 3.4.12. When pages were not found it generated a nice standard error page (https://www.dropbox.com/s/udevmgjwra0p4zf/Screenshot%202018-06-27%2022.12.27.png?dl=0) but now an 500 Internal Server error is returned and the logs show

```
[critical] Uncaught Exception: Unable to find the controller for path "/non_existing". The route is wrongly configured.
```

**Comment by nicolas-grekas** [comment]
You can register an exception listener instead by the way.
